<a href="https://colab.research.google.com/github/jcjimenezb123/MetodosNumericosPython/blob/master/Proporcion_aurea.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Proporción aurea
---
La proporción áurea, es un concepto matemático y estético que ha fascinado a los artistas, arquitectos y matemáticos a lo largo de la historia. Se denota generalmente por la letra griega φ (phi).

La razón dorada se define en términos de una proporción entre dos segmentos, de manera que el cociente de la longitud total a la longitud del segmento más largo es igual al cociente de la longitud del segmento más largo a la longitud del segmento más corto. Matemáticamente, se puede expresar de la siguiente manera:
$$
(a + b) / a = a / b = φ
$$
Donde $a$ representa la longitud del segmento más largo, $b$ representa la longitud del segmento más corto y $φ$ es la razón dorada.

|           a    | b |
|----|---|
|<------------------->|<------>|

La ecuación anterior la podemos escribir como
$$
a/a + b / a = a / b = φ
$$
$$
1 + 1 / φ = φ
$$
multiplicando por $φ$
$$
φ + 1 = φ^2
$$

$$
φ^2 - φ - 1 =0
$$

aplicando la regla general para obtener $φ$

$$
φ = \frac{\sqrt{5}-1}{2} \approx 0.6180339887498949
$$

In [2]:
import numpy as np
phi = (np.sqrt(5)-1)/2
print(phi,1/phi)

0.6180339887498949 1.6180339887498947


[Proporcion aurea](https://ovacen.com/proporcion-aurea-que-es/)

Secuencia de Fibonacci
---
La serie de Fibonacci es una secuencia numérica que comienza con los números 0 y 1, y luego cada número subsiguiente es la suma de los dos números anteriores. La secuencia se extiende infinitamente y se denota comúnmente como F(n), donde "n" es la posición en la secuencia.

La serie de Fibonacci se construye de la siguiente manera:

1. Los dos primeros números de la secuencia son 0 y 1: F(0) = 0 y F(1) = 1.

1. A partir del tercer número en adelante, cada número en la secuencia es la suma de los dos números anteriores: F(n) = F(n-1) + F(n-2).

Por lo tanto, los primeros números de la secuencia de Fibonacci son: 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, y así sucesivamente.

La serie de Fibonacci tiene muchas propiedades interesantes y se encuentra en diversas áreas de las matemáticas y la naturaleza. Algunas de las propiedades notables incluyen:

1. La razón entre dos números consecutivos en la serie de Fibonacci se acerca cada vez más a la razón dorada (φ ≈ 1.618033988749895) a medida que avanzamos en la secuencia. Esta relación es conocida como la aproximación de la razón dorada y es una propiedad fascinante y notable.

1. La serie de Fibonacci está estrechamente relacionada con la espiral de Fibonacci, una espiral que se forma conectando arcos que tienen longitudes proporcionales a los números de Fibonacci.

1. Los números de Fibonacci tienen aplicaciones en matemáticas, ciencias de la computación, biología y otras áreas. Aparecen en problemas relacionados con combinatoria, teoría de números, algoritmos y muchos otros campos.

La serie de Fibonacci es un ejemplo clásico de una secuencia recursiva y ha cautivado a matemáticos y entusiastas durante siglos debido a sus patrones intrigantes y su presencia en la naturaleza y el arte.

In [ ]:
fibonacci=[] #<--- valores iniciales de la secuencia de Fibonacci

fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)
fibonacci.append(fibonacci[-2]+fibonacci[-1]);print(fibonacci)

print(fibonacci[-2]/fibonacci[-1])

[Fibonacci sequence video](https://www.youtube.com/watch?v=P0tLbl5LrJ8)

[Fibonacci music](https://www.youtube.com/watch?v=IGJeGOw8TzQ)

[Literalus video](https://www.youtube.com/watch?v=uOHkeH2VaE0&t=50s)